### PR TITLE
Add enhancement proposal process

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This PR gives us a public place to store our design proposal as well as a structure. This also adds transparency and allows the community to participate.

Based on https://github.com/kubernetes/enhancements

### CODE REVIEW

**Review by rzetelskik — APPROVED** [review]
Looks good, one nit. Maybe we could also use a document describing the process itself (as in https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/0000-kep-process)?

File: enhancements/proposal-template.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+What are the guarantees? Make sure this is in the test plan.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+Why should this KEP __not__ be implemented?
```
[the remark this answers is not in the document]
Comment (tnozicka) [reply enhancements/proposal-template.md]: fixed, thx

**Comment by tnozicka** [comment]
> Maybe we could also use a document describing the process itself (as in https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/0000-kep-process)?

We do not have it coupled to the other processes as kube does, so I think we should be good just with the template and pointing people to it for now. (We may reconsider with scale.)

**Review by zimnx — APPROVED** [review]
LGTM, I hope opening for community will bring us more contributions.

**Comment by zimnx** [comment]
Please rebase

**Comment by tnozicka** [comment]
(clean rebase)

**Review by zimnx — APPROVED** [review]

**Pull request merged**